temporalio/client.py

```diff
@@ -0,0 +1,935 @@
+"""Client for accessing Temporal."""
+
+import logging
+import os
+import socket
+import uuid
+from dataclasses import dataclass
+from datetime import timedelta
+from enum import IntEnum
+from typing import Any, Generic, Iterable, Mapping, Optional, TypeVar, Union, cast
+
+import temporalio.api.common.v1
+import temporalio.api.enums.v1
+import temporalio.api.failure.v1
+import temporalio.api.history.v1
+import temporalio.api.taskqueue.v1
+import temporalio.api.workflowservice.v1
+import temporalio.common
+import temporalio.converter
+import temporalio.failure
+import temporalio.workflow_service
+
+logger = logging.getLogger(__name__)
+
+
+class WorkflowIDReusePolicy(IntEnum):
+    """How already-in-use workflow IDs are handled on start.
+
+    See :py:class:`temporalio.api.enums.v1.WorkflowIdReusePolicy`.
+    """
+
+    ALLOW_DUPLICATE = int(
+        temporalio.api.enums.v1.WorkflowIdReusePolicy.WORKFLOW_ID_REUSE_POLICY_ALLOW_DUPLICATE
+    )
+    """See :py:attr:`temporalio.api.enums.v1.WorkflowIdReusePolicy.WORKFLOW_ID_REUSE_POLICY_ALLOW_DUPLICATE`."""
+
+    ALLOW_DUPLICATE_FAILED_ONLY = int(
+        temporalio.api.enums.v1.WorkflowIdReusePolicy.WORKFLOW_ID_REUSE_POLICY_ALLOW_DUPLICATE_FAILED_ONLY
+    )
+    """See :py:attr:`temporalio.api.enums.v1.WorkflowIdReusePolicy.WORKFLOW_ID_REUSE_POLICY_ALLOW_DUPLICATE_FAILED_ONLY`."""
+
+    REJECT_DUPLICATE = int(
+        temporalio.api.enums.v1.WorkflowIdReusePolicy.WORKFLOW_ID_REUSE_POLICY_REJECT_DUPLICATE
+    )
+    """See :py:attr:`temporalio.api.enums.v1.WorkflowIdReusePolicy.WORKFLOW_ID_REUSE_POLICY_REJECT_DUPLICATE`."""
+
+
+class WorkflowQueryRejectCondition(IntEnum):
```

Comment:
Same

Reply:
Same as in it should be in `common`? Why? Can you give more reasoning on this stuff? Why is it not client-specific in TypeScript? It's only used in the client, why would we take it out of the client?

Reply:
If this is the reason to put it in common, should I open a TypeScript issue to move `WorkflowQueryInput` (for use by workflow interceptors) and `QueryRejectedError` over too?

Or is there another reason to put it in common? Go puts the query option that includes rejection [in the client area](https://pkg.go.dev/go.temporal.io/sdk/client#QueryWorkflowWithOptionsRequest) (it reuses the proto enum though). Java reuses the proto one too.

I think TS is the only one that considers query reject condition common code.

Reply:
Eventually we would probably want to move everything query related into common.
It's not urgent but it's more "correct".
Even if you use an activity for querying a workflow from another workflow you still might want to pass the reject condition from the workflow.

Reply:
After discussion, we should move this to common and move `QueryRejectedError` to `exceptions`.

Reply:
After further thought, `WorkflowQueryRejectedError` cannot move to common exceptions place due to it having a workflow execution status which means it wouldn't be applicable to activities

Reply:
Moved `WorkflowQueryRejectCondition`